airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -43,7 +45,7 @@ def __init__(self, stream_instance_map: Mapping[str, Stream], state: Union[List[
             )
         self.per_stream_states = per_stream_states
 
-    def get_stream_state(self, stream_name: str, namespace: Optional[str]) -> Mapping[str, Any]:
+    def get_stream_state(self, stream_name: str, namespace: Optional[str]) -> MutableMapping[str, Any]:
```

Comment:
Why do we need to return a `MutableMapping` here? Isn't `Mapping` a subset of the `MutableMapping` interface?

Reply:
because users (AbstractSource and Stream.get_updated_state) expect the stream_state to be mutable